### ISSUE TITLE
Custom Annotations with pipes confound VCF CSQ output format

### ISSUE THREAD

**Issue by james-lawlor** [Ensembl/ensembl-vep#93]
Hi,
TLDR: Custom annotations should convert pipes "|" into ampersands "&" to preserve VCF format. 

I've discovered an issue with VEP version 90.5 custom annotations: If you have VEP include a VCF INFO field annotation which includes pipes, such as the "DP_HIST_ALT" field from the gnomAD exome VCF, it is placed verbatim into VEP's CSQ field, where the extra pipes cause a mis-match with the key in the header. For example:

Input: 
```1       12266   .       G       A       52.43   VQSRTrancheSNP99.90to100.00     AC=1    GT:AD:DP:GQ:PGT:PID:PL  ./.:0,0:0```

Custom annotations selected from gnomAD in order:
```
AC_raw=51;
DP_HIST_ALT=33|3|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;AB_HIST_ALT=0|0|0|0|0|2|0|0|2|0|9|0|0|8|0|0|0|0|0|0;
AN_raw=11524;
```

Output header describing CSQ line:
```
##INFO=<ID=CSQ,Number=.,Type=String,Description="Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|SYMBOL_SOURCE|HGNC_ID|SOURCE|gnomadEx|gnomadEx_AC_raw|gnomadEx_DP_HIST_ALT|gnomadEx_AN_raw">
##INFO=<ID=gnomadEx,Number=.,Type=String,Description="/gpfs/gpfs2/reference/vep/custom/GRCh37/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz (exact)">
##INFO=<ID=gnomadEx_AC_raw,Number=.,Type=String,Description="AC_raw field from /gpfs/gpfs2/reference/vep/custom/GRCh37/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz">
##INFO=<ID=gnomadEx_DP_HIST_ALT,Number=.,Type=String,Description="DP_HIST_ALT field from /gpfs/gpfs2/reference/vep/custom/GRCh37/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz">
##INFO=<ID=gnomadEx_AN_raw,Number=.,Type=String,Description="AN_raw field from /gpfs/gpfs2/reference/vep/custom/GRCh37/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz">
```

Output VCF line CSQ: 
```
CSQ=A|intergenic_variant|MODIFIER||||||||||||||||||||||1:12266-12266|51|33|3|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|11524
```

This makes interpreting the output CSQ line very difficult as it no longer matches the header documentation and has an unpredictable number of output fields. It would make sense to convert the pipes into ampersands when processing the custom annotation, much like the case where a custom annotation includes commas. 

Notes:
1. gnomAD exome VCF downloaded from `ftp://ftp.ensembl.org/pub/data_files/homo_sapiens/GRCh37/variation_genotype/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz`
2. VEP version
```
Versions:
  ensembl              : 90.4a44397
  ensembl-funcgen      : 90.d61fff0
  ensembl-io           : 90.9a148ea
  ensembl-variation    : 90.58bf949
  ensembl-vep          : 90.5
```
3. VEP command: `vep --config test.ini -i test_case_input.vcf -o test_case_output.vcf --force_overwrite`
4. Options from test.ini:
```
custom /gpfs/gpfs2/reference/vep/custom/GRCh37/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz,gnomadEx,vcf,exact,0,AC_raw,DP_HIST_ALT,AN_raw
format vcf
vcf TRUE
```
5. Attached files (as text files)
[test.ini.txt](https://github.com/Ensembl/ensembl-vep/files/1443395/test.ini.txt)
[test_case_input.vcf.txt](https://github.com/Ensembl/ensembl-vep/files/1443402/test_case_input.vcf.txt)
[test_case_output.vcf.txt](https://github.com/Ensembl/ensembl-vep/files/1443403/test_case_output.vcf.txt)

6. relevant line of gnomad exome VCF:
```
tabix /gpfs/gpfs2/reference/vep/custom/GRCh37/gnomad.exomes.r2.0.1.sites.noVEP.vcf.gz 1:12266-12266
1	12266	.	G	A	2721.48	AC0	AC=0;AF=.;AN=0;BaseQRankSum=8.04000e-01;ClippingRankSum=0.00000e+00;DP=18372;FS=0.00000e+00;InbreedingCoeff=-1.37600e-01;MQ=2.23200e+01;MQRankSum=-7.27000e-01;QD=1.61000e+01;ReadPosRankSum=4.06000e-01;SOR=2.03000e-01;VQSLOD=1.27000e+00;VQSR_culprit=MQ;GQ_HIST_ALT=0|15|0|0|0|2|0|2|5|0|1|1|1|3|0|6|0|0|0|0;DP_HIST_ALT=33|3|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;AB_HIST_ALT=0|0|0|0|0|2|0|0|2|0|9|0|0|8|0|0|0|0|0|0;GQ_HIST_ALL=3755|1684|188|101|11|4|0|2|5|0|1|1|1|3|0|6|0|0|0|0;DP_HIST_ALL=5598|163|1|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0|0;AB_HIST_ALL=0|0|0|0|0|2|0|0|2|0|9|0|0|8|0|0|0|0|0|0;AC_AFR=0;AC_AMR=0;AC_ASJ=0;AC_EAS=0;AC_FIN=0;AC_NFE=0;AC_OTH=0;AC_SAS=0;AC_Male=0;AC_Female=0;AN_AFR=0;AN_AMR=0;AN_ASJ=0;AN_EAS=0;AN_FIN=0;AN_NFE=0;AN_OTH=0;AN_SAS=0;AN_Male=0;AN_Female=0;AF_AFR=.;AF_AMR=.;AF_ASJ=.;AF_EAS=.;AF_FIN=.;AF_NFE=.;AF_OTH=.;AF_SAS=.;AF_Male=.;AF_Female=.;GC_AFR=0,0,0;GC_AMR=0,0,0;GC_ASJ=0,0,0;GC_EAS=0,0,0;GC_FIN=0,0,0;GC_NFE=0,0,0;GC_OTH=0,0,0;GC_SAS=0,0,0;GC_Male=0,0,0;GC_Female=0,0,0;AC_raw=51;AN_raw=11524;AF_raw=4.42555e-03;GC_raw=5726,21,15;GC=0,0,0;Hom_AFR=0;Hom_AMR=0;Hom_ASJ=0;Hom_EAS=0;Hom_FIN=0;Hom_NFE=0;Hom_OTH=0;Hom_SAS=0;Hom_Male=0;Hom_Female=0;Hom_raw=15;Hom=0;POPMAX=.;AC_POPMAX=.;AN_POPMAX=.;AF_POPMAX=.;DP_MEDIAN=3;DREF_MEDIAN=1.18961e-08;GQ_MEDIAN=39;AB_MEDIAN=5.00000e-01;AS_RF=2.37241e-01;AS_FilterStatus=AC0
```
7. Edited to re-anonymize sample identifier, just in case. 

Thank you!
James

**Comment by willmclaren** [comment]
Thanks for the super-detailed report! Now fixed in release/90.7
